libs/libc/unistd/lib_getcwd.c

```diff
@@ -127,3 +127,42 @@ FAR char *getcwd(FAR char *buf, size_t size)
   strlcpy(buf, pwd, size);
   return buf;
 }
+
+/****************************************************************************
+ * Name: get_current_dir_name
+ *
+ * Description:
+ *   get_current_dir_name() function will malloc an array big enough to
+ *   hold the absolute pathname of the current working directory.  If the
+ *   environment variable PWD is set, and its value is correct, then
+ *   that value will be returned. The caller should free the returned
+ *   buffer.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   On success, get_current_dir_name() returns a pointer to a string
+ *   containing the pathname of the current working directory.
+ *   Otherwise, get_current_dir_name() returns a null pointer and
+ *   sets errno to indicate the error:
+ *
+ *   EACCES
+ *     Read or search permission was denied for a component of the pathname.
+ *   ENOMEM
+ *     Insufficient storage space is available.
+ *
```

Comment:
this seems to contain a copy-and-paste from linux man page. https://linux.die.net/man/3/get_current_dir_name
is it copyright-wise ok?

Reply:
> this seems to contain a copy-and-paste from linux man page. https://linux.die.net/man/3/get_current_dir_name is it copyright-wise ok?

the comments is same as "getcwd(FAR char *buf, size_t size)", not copied from " https://linux.die.net/man/3/get_current_dir_name"

Reply:
> > this seems to contain a copy-and-paste from linux man page. https://linux.die.net/man/3/get_current_dir_name is it copyright-wise ok?
> 
> the comments is same as "getcwd(FAR char *buf, size_t size)", not copied from " https://linux.die.net/man/3/get_current_dir_name"

sorry, what do you mean?
nuttx lib_getcwd.c doesn't seem to have the same comments.